go/src/meguca/server/router.go

```diff
@@ -257,18 +257,38 @@ func youTubeData(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	video, err := info.GetDownloadURL(info.Formats.
+	vidFormat := info.Formats.
 		Filter(ytdl.FormatExtensionKey, []interface{}{"webm"}).
-		Filter(ytdl.FormatResolutionKey, []interface{}{"360p"})[0])
+		Filter(ytdl.FormatResolutionKey, []interface{}{"360p"})[0]
```

Comment:
Chiru, do you know what's an out of bounds read? How do you know that this function always returns a non-empty array? You need to check it's length before you [0] it.